ui/src/components/Publication/Creation/MainText.tsx

```diff
@@ -161,49 +159,49 @@ const MainText: React.FC = (): React.ReactElement | null => {
                 continue;
             }
 
-            // Pattern using regex groups to retrieve reference text and to determine whether a string contains DOI or URI
-            const pattern =
-                /(?<TEXT>.+?(?=http))((?<DOI>((((http|https):\/\/)(([-a-zA-Z0-9_]{1,265}([^\s]+)))))(10{1}\.([^\n]+)))|(?<URL>((((http|https):\/\/)(([-a-zA-Z0-9_]{1,265}([^\n]+)))))))|(?<TEXTONLY>.+?(?=$))/g;
-
-            const matches = textContent.matchAll(pattern);
-
-            // Iterate through matches
-            if (matches) {
-                for (const match of matches) {
-                    let location: string | null;
-                    let type: Interfaces.ReferenceType;
-                    let text = currentParagraph; // original html string
-
-                    if (match?.groups?.DOI) {
-                        type = 'DOI';
-                        location = match.groups.DOI;
-                        text = text.replace(
-                            match.groups.DOI,
-                            `<a href="${match.groups.DOI}" target="_blank" rel="noreferrer noopener">${match.groups.DOI}</a>`
-                        );
-                    } else if (match?.groups?.URL) {
-                        type = 'URL';
-                        location = match.groups.URL;
-                        text = text.replace(
-                            match.groups.URL,
-                            `<a href="${match.groups.URL}" target="_blank" rel="noreferrer noopener">${match.groups.URL}</a>`
-                        );
-                    } else {
-                        type = 'TEXT';
-                        location = null;
-                    }
-
-                    const newReference: Interfaces.Reference = {
-                        id: cuid(),
-                        publicationId,
-                        type,
-                        text,
-                        location
-                    };
+            const URL_REGEX = /(http|ftp|https):\/\/([\w_-]+(?:(?:\.[\w_-]+)+))([\w.,@?^=%&:\/~+#-]*[\w@?^=%&\/~+#-])/g;
+
+            const urlMatches = textContent.match(URL_REGEX) || [];
+            if (!urlMatches?.length) {
+                // no urls have been found
+                referencesArray.push({
+                    id: cuid(),
+                    publicationId,
+                    type: 'TEXT',
+                    text: currentParagraph,
+                    location: null
+                });
+                // skip the other checks
+                continue;
+            }
 
-                    referencesArray.push(newReference);
-                }
+            // reverse the order for easier access
+            const reversedUrlMatches = urlMatches.reverse(); // we only need the last DOI/URL
+
+            // get the last valid DOI url
+            const DOI_REGEX = /(10\.[0-9a-zA-Z]+\/(?:(?!["&\'])\S)+)\b/g;
```

Comment:
we just discussed in standup we know this fails in some case still and it likely a regex in itself can't match everything so we will revisit this in the DOI task, for now I'm happy to merge this

Reply:
Yes, there are some edge cases that we will try to fix in [OCT-352](https://jiscdev.atlassian.net/browse/OCT-352)